### PR TITLE
Hy wiring everything up

### LINKED ISSUE #14: Wiring everything up
#### As a developer I want to be able to record the feedback collected in the backend.

__Story__:

You now have a working form that allows users to write up their feedback. The last step is to send an API request to record it. We will not be implement any handling of the request other than the required routes, controllers, and actions for it to function.

Happy coding!

__Acceptance criteria__:
- [ ] There is a new PostFeedbackService service object that uses the `post` method export from `utils/helpers.js` to send an API request with the form data (POST body structure is up to you)
- [ ] There is a new API controller (with namespace) `Api::FeedbacksController` that has an action to handle the feedback request, which always returns a 200 status code with no content
- [ ] The path that accepts the request is `/api/feedbacks`
- [ ] When the user submits the request, a message is displayed stating that it was successful

__Discussion Topics__:
- [ ] How might you be able to provide the form state to the PostFeedbackService that uses it to send the request?

__References__:
* https://reactjs.org/tutorial/tutorial.html
* https://mobx.js.org/getting-started.html
* https://sinonjs.org/

### PR DESCRIPTION
**Pull request by HongquanYu**
Closes #14 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [x] I have replaced `X` above with the associated issue number.
- [x] I have changed the issue's label to "pull request".
- [x] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.
- [x] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by HongquanYu** [comment]
[Wiring everything up](https://trello.com/c/LHULpNtq/392-wiring-everything-up)

File: app/javascript/image_sharing/components/App.js [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+import { inject, observer } from 'mobx-react';
+import PropTypes from 'prop-types';
+
 import Header from './Header';
 import Footer from "./Footer";
 import FeedbackForm from "./FeedbackForm";
 import FeedbackList from "./FeedbackList";
-import PropTypes from 'prop-types';
 
+@observer
```
Comment (almgong) [inline app/javascript/image_sharing/components/App.js]: Do we need to inject `stores` for this component?

File: app/javascript/image_sharing/components/App.js [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
 class App extends Component {
   static propTypes = {
     stores: PropTypes.object.isRequired
   };
 
   render() {
-    const store = this.props.stores.feedbackStore;
+    const flashMsg = this.props.stores.flashMsgStore.flashMessage;
```
Comment (almgong) [inline app/javascript/image_sharing/components/App.js]: This works just fine, but as a tip, you will commonly see `UIStateStore`s in other React apps in our codebases. Their purpose is to store all the UI-based (application specific) data, such as the contents of the flash message alert. Here is a reference: https://mobx.js.org/best/store.html#stores-for-the-user-interface-state.

File: package.json [hunk 3]
```diff
@@ -12,10 +12,12 @@
     "coffeescript": "1.12.7",
     "mobx": "^5.9.0",
     "mobx-react": "^5.4.3",
+    "production": "0.0.2",
```
Comment (almgong) [inline package.json]: What is this package used for?

File: package.json [hunk 4]
```diff
@@ -12,10 +12,12 @@
...
     "prop-types": "^15.6.1",
     "react": "^16.8.4",
     "react-dom": "^16.8.4",
-    "reactstrap": "^7.1.0"
+    "reactstrap": "^7.1.0",
+    "whatwg-fetch": "^3.0.0"
```
Comment (almgong) [inline package.json]: Is this used anywhere?

File: package.json [hunk 5]
```diff
@@ -21,7 +22,12 @@
   },
   "devDependencies": {
     "assert": "^1.4.1",
+    "babel-core": "^6.26.3",
     "babel-eslint": "^10.0.1",
+    "babel-loader": "^8.0.6",
+    "babel-polyfill": "^6.26.0",
```
Comment (almgong) [inline package.json]: What are all of these changes for?

File: app/javascript/image_sharing/test/FeedbackForm.spec.js [hunk 6]
```diff
@@ -24,14 +31,34 @@ describe('<FeedbackForm />', () => {
...
+    expect(stores.feedbackStore.addFeedback.withArgs('test name 2', 'test comment 2').calledOnce).to.equal(true);
+    expect(wrapper.instance().feedback.name).to.equal('test name 2');
+    expect(wrapper.instance().feedback.comment).to.equal('test comment 2');
+  });
+
+  it('should display successful flash message when submit successfully', async () => {
+    const wrapper = shallow(<FeedbackForm.wrappedComponent stores={stores} />);
+    wrapper.instance().feedback = {name: 'test name', comment: 'test comment'};
+
+    await wrapper.instance().onFormSubmit();
```
Comment (almgong) [inline app/javascript/image_sharing/test/FeedbackForm.spec.js]: Should we add a test to check where the `onFormSubmit` is passed as a component prop?

File: package.json (hunk 4 above)
Comment (HongquanYu) [reply package.json]: This is used in utils/helper.js. The first line has imported "whatwg-fetch"

File: app/javascript/image_sharing/components/App.js (hunk 1 above)
Comment (HongquanYu) [reply app/javascript/image_sharing/components/App.js]: Displaying status message uses stores here in App.js

File: app/javascript/image_sharing/components/App.js (hunk 2 above)
Comment (HongquanYu) [reply app/javascript/image_sharing/components/App.js]: Thanks!

File: package.json (hunk 3 above)
Comment (HongquanYu) [reply package.json]: deleted

File: package.json (hunk 5 above)
Comment (HongquanYu) [reply package.json]: The tests use async/await which requires those, otherwise throwing `regeneratorRuntime is not defined.` error

**Pull request merged**